Common/Data/Custom/PsychSignal/PsychSignalSentimentData.cs

```diff
@@ -0,0 +1,133 @@
+
+        /// <summary>
+        /// Total bullish scored messages
+        /// </summary>
+        public int BullScoredMessages;
+
+        /// <summary>
+        /// Total bearish scored messages
+        /// </summary>
+        public int BearScoredMessages;
+
+        /// <summary>
+        /// Bull/Bear message ratio.
+        /// </summary>
+        /// <remarks>If bearish messages equals zero, then the resulting value equals zero</remarks>
+        public decimal BullBearMessageRatio;
+
+        /// <summary>
+        /// Total messages scanned.
+        /// </summary>
+        /// <remarks>
+        /// Sometimes, there will be no bull/bear rated messages, but nonetheless had messages scanned.
+        /// This field describes the total fields that were scanned in a minute
+        /// </remarks>
+        public int TotalScoredMessages;
+
+        /// <summary>
+        /// Retrieve Psychsignal data from disk and return it to user's custom data subscription
+        /// </summary>
+        /// <param name="config">Configuration</param>
+        /// <param name="date">Date of this source file</param>
+        /// <param name="isLiveMode">true if we're in livemode, false for backtesting mode</param>
+        /// <returns></returns>
+        public override SubscriptionDataSource GetSource(SubscriptionDataConfig config, DateTime date, bool isLiveMode)
+        {
+            return new SubscriptionDataSource(
+                Path.Combine(
+                    Globals.DataFolder,
+                    "equity",
+                    config.Market,
+                    "alternative",
+                    "psychsignal",
+                    $"{config.Symbol.Value.ToLower()}.zip#{date:yyyyMMdd}.csv"
+                ),
+                SubscriptionTransportMedium.LocalFile,
+                FileFormat.Csv
+            );
+        }
+
+        /// <summary>
+        /// Reads a single entry from psychsignal's data source.
+        /// </summary>
+        /// <param name="config">Subscription data config setup object</param>
+        /// <param name="line">Line of the source document</param>
+        /// <param name="date">Date of the requested data</param>
+        /// <param name="isLiveMode">true if we're in live mode, false for backtesting mode</param>
+        /// <returns>
+        ///     Instance of the T:BaseData object containing psychsignal specific data
+        /// </returns> 
+        public override BaseData Reader(SubscriptionDataConfig config, string line, DateTime date, bool isLiveMode)
+        {
+            try
+            {
+                var csv = line.Split(',');
+
+                var timestamp = new DateTime(date.Year, date.Month, date.Day).AddMilliseconds(
+                    Convert.ToDouble(csv[0])
+                );
+                var bullIntensity = Convert.ToDecimal(csv[1]);
+                var bearIntensity = Convert.ToDecimal(csv[2]);
+                var bullMinusBear = Convert.ToDecimal(csv[3]);
```

Comment:
I think these conversions should be changed to `Convert.ToDecimal(..., CultureInfo.InvariantCulture)` for international users.